internal/exec/resolvable/resolvable.go

```diff
@@ -161,10 +158,50 @@ func ApplyResolver(s *types.Schema, resolver interface{}, registeredDirectives m
 	}, nil
 }
 
+func applyDirectives(s *types.Schema, visitors []directives.Directive) (map[string]directives.Directive, error) {
+	byName := make(map[string]directives.Directive, len(s.Directives))
+
+	for name := range s.Directives {
+		for _, v := range visitors {
+			if !v.ImplementsDirective(name) {
+				continue
+			}
+
+			if existing, ok := byName[name]; ok {
+				return nil, fmt.Errorf("multiple implementations registered for directive %q. Implementation types %T and %T", name, existing, v)
+			}
+
+			byName[name] = v
+		}
+
+		if _, ok := byName[name]; !ok {
+			if name == "include" || name == "skip" || name == "deprecated" || name == "specifiedBy" {
+				// Special case directives, ignore
+				continue
+			}
+
+			return nil, fmt.Errorf("no visitors have been registered for directive %q", name)
+		}
+	}
+
+	// Validate that all the directive visitors are valid; each must implement *at least 1* of the optional functions
+	for name, v := range byName {
+		// At least 1 of the optional directive functions must be defined for each directive.
+		// For now this is the only valid directive function
+		if _, ok := v.(directives.ResolverInterceptor); ok {
+			continue
```

Comment:
I want the happy path to be last. So this must change to:
```go
if !ok {
   return nil, fmt.Errorf(...)
}
```
No need to use `continue` here, please.

Reply:
The intent here was to handle each of the supported interface types, so break early on mismatch wasn't going to be possible once validation directives are added (unless I'm misunderstanding an alternative here)

Reply:
One way to achieve this would be:
```go
	for name, v := range byName {
		if _, ok := v.(directives.ResolverInterceptor); !ok {
			return nil, fmt.Errorf("directive %q (implemented by %T) does not implement a valid directive visitor function", name, v)
		}
	}

	return byName, nil
```
note that `continue` is not used in the above example. 
Another approach would be:
```go
	for name, v := range byName {
		// At least 1 of the optional directive functions must be defined for each directive.
		// For now this is the only valid directive function
		switch t := v.(type) {
		case directives.ResolverInterceptor:
		// case directives.Validator:
		default:
			return nil, fmt.Errorf("directive %q (implemented by %T) does not implement a valid directive visitor function", name, v)
		}
	}

	return byName, nil
```